features/steps/dataset_feature.go

```diff
@@ -0,0 +1,240 @@
+	"github.com/ONSdigital/dp-dataset-api/mongo"
+	"github.com/ONSdigital/dp-dataset-api/service"
+	serviceMock "github.com/ONSdigital/dp-dataset-api/service/mock"
+	"github.com/ONSdigital/dp-dataset-api/store"
+	storeMock "github.com/ONSdigital/dp-dataset-api/store/datastoretest"
+	"github.com/ONSdigital/dp-healthcheck/healthcheck"
+	kafka "github.com/ONSdigital/dp-kafka/v2"
+	"github.com/ONSdigital/dp-kafka/v2/kafkatest"
+	featuretest "github.com/armakuni/dp-go-featuretest"
+	"github.com/benweissmann/memongo"
+	"github.com/cucumber/godog"
+	"github.com/globalsign/mgo"
+	"github.com/maxcnunes/httpfake"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+type DatasetFeature struct {
+	ErrorFeature    featuretest.ErrorFeature
+	svc             *service.Service
+	errorChan       chan error
+	Datasets        []*models.Dataset
+	MongoClient     *mongo.Mongo
+	Config          *config.Configuration
+	HTTPServer      *http.Server
+	FakeAuthService *httpfake.HTTPFake
+	ServiceRunning  bool
+}
+
+func NewDatasetFeature(mongoCapability *featuretest.MongoCapability) *DatasetFeature {
+
+	f := &DatasetFeature{
+		HTTPServer:      &http.Server{},
+		errorChan:       make(chan error),
+		Datasets:        make([]*models.Dataset, 0),
+		FakeAuthService: httpfake.New(),
+		ServiceRunning:  false,
+	}
+
+	var err error
+
+	f.Config, err = config.Get()
+	if err != nil {
+		panic(err)
+	}
+
+	f.Config.ZebedeeURL = f.FakeAuthService.ResolveURL("")
+
+	mongodb := &mongo.Mongo{
+		CodeListURL: "",
+		Collection:  "datasets",
+		Database:    memongo.RandomDatabase(),
+		DatasetURL:  "datasets",
+		URI:         mongoCapability.Server.URI(),
+	}
+
+	if err := mongodb.Init(); err != nil {
+		panic(err)
+	}
+
+	f.MongoClient = mongodb
+
+	initMock := &serviceMock.InitialiserMock{
+		DoGetMongoDBFunc:       f.DoGetMongoDB,
+		DoGetGraphDBFunc:       f.DoGetGraphDBOk,
+		DoGetKafkaProducerFunc: f.DoGetKafkaProducerOk,
+		DoGetHealthCheckFunc:   f.DoGetHealthcheckOk,
+		DoGetHTTPServerFunc:    f.DoGetHTTPServer,
+	}
+
+	f.svc = service.New(f.Config, service.NewServiceList(initMock))
+
+	return f
+}
+
+func (f *DatasetFeature) RegisterSteps(ctx *godog.ScenarioContext) {
+	ctx.Step(`^private endpoints are enabled$`, f.PrivateEndpointsAreEnabled)
+	ctx.Step(`^I am not identified$`, f.IAmNotIdentified)
+	ctx.Step(`^I am identified as "([^"]*)"$`, f.IAmIdentifiedAs)
+	ctx.Step(`^I have these datasets:$`, f.IHaveTheseDatasets)
+	ctx.Step(`^the document in the database for id "([^"]*)" should be:$`, f.TheDocumentInTheDatabaseForIdShouldBe)
+}
+
+func (f *DatasetFeature) Reset() *DatasetFeature {
+	f.Datasets = make([]*models.Dataset, 0)
+	f.MongoClient.Database = memongo.RandomDatabase()
+	f.MongoClient.Init()
+	f.Config.EnablePrivateEndpoints = false
+	f.FakeAuthService.Reset()
+	return f
+}
+
+func (f *DatasetFeature) Close() error {
+	if f.svc != nil && f.ServiceRunning {
+		f.svc.Close(context.Background())
```

Comment:
Better to pass context value into method in case calling Close is slow, that way we can force a shutdown/stop using context timeouts

Reply:
Also we should be handling the error returned from Close function - always handle the error and pass back up the chain

Reply:
@nshumoogum given this method is called at end of the feature tests, which context do you propose to use?